keps/sig-api-machinery/5000-api-linting-crd-schema-tooling/README.md

```diff
@@ -265,20 +266,62 @@ Currently, it is proposed that:
+| scope | CRD | Validates that scope doesn't change for the CRD | Yes | Yes |
+| existingfieldremoval | CRD | Validates that existing fields are not removed from the CRD | Yes | Yes |
+| storedversionremoval | CRD | Validates that stored versions are not removed from the CRD. Only valid for CRDs that are previously present on a cluster | Yes | Yes |
+| format | Property | Validates compatibility of changes to the format of a property. Changes to property formats are flagged | No | Yes |
+| exclusiveMaximum | Property | Validates compatibility of changes to the exclusiveMaximum of a property. Net new exclusiveMaximum constraints or a decrease in exclusiveMaximum are flagged | No | Yes |
+| exclusiveMinimum | Property | Validates compatibility of changes to the exclusiveMinimum of a property. Net new exclusiveMinimum constraints or an increase in exclusiveMinimum are flagged | No | Yes |
+| uniqueItems | Property | Validates compatibility of changes to the uniqueItems of a property. Net new uniqueItems constraints are flagged | No | Yes | 
+| pattern | Property | Validates compatibility of changes to the pattern constraint of a property. Net new pattern constraints and more restrictive validations are flagged | No | Yes |
+| nullable | Property  | Validates compatibility of changes to the nullable constraint of a property. Making a property no longer nullable is flagged | No | Yes |
+| multipleOf | Property | Validates compatibility of changes to the multipleOf constraint of a property. Changes to the multipleOf constraint are flagged | No | Yes |
+| allOf | Property | Validates compatibility of changes to the allOf constraint of a property. Net new and addition of allOf constraints are flagged | No | Yes |
+| oneOf | Property | Validates compatibility of changes to the oneOf constraint of a property. Net new and removal of oneOf constraints are flagged | No | Yes |
+| anyOf | Property | Validates compatibility of changes to the anyOf constraint of a property. Net new and removal of anyOf constraints are flagged | No | Yes |
+| not | Property | Validates compatibility of changes to the not constraint of a property. Net new not constraints are flagged | No | Yes |
+
+Proposed validations are tracked via https://github.com/everettraven/crd-diff/issues/3.
+
+Validations in https://github.com/openshift/crd-schema-checker  both existing and proposed:
+
+| Name | Scope | Description | Implemented | Carry |
+| ---- | ----- | ----------- | ----------- | ----- |
+| NoBools | Property | Validates that CRD properties are not of type `boolean` | Yes | No, if desired this is better suited for a linter in `kal` |
+| NoFloats | Property | Validates that CRD properties are not of type `number` | Yes | No, if desired this is better suited for a linter in `kal` |
+| NoUints | Property | Validates that CRD properties are not of type `uint` | Yes | No, if desired this is better suited for a linter in `kal` |
```

Comment:
Already implemented in KAL as part of the `integers` check, that also checks you don't have `int` or `int8` or `int16` which are also discouraged

Reply:
Updated in https://github.com/kubernetes/enhancements/pull/5102/commits/e69b3eda1028594e6d76b513e3183e12bc8aee91